File: swig/python/mktable.py
import greendb
import sys

def mktable(args):
	table_name = args[0];

	env = greendb.GreenEnv(".")
	env.open()
	
	table = greendb.Table(table_name, env)
	schema = table.get_schema()
	size = schema.size()
	if (size != 0):
		print size, "table already created"
		return
	args = args[1:]	
	i = 0
	typemap = greendb.TypeMap.get_type_map()
	while i < len(args):
		type = typemap.get_type_id(args[i])
		i = i + 1
		name = args[i]
		i = i + 1
		print "adding ", type, name
		schema.add_column(name, type, 0)
	
	table.close()
	env.close()

if __name__ == '__main__':
	mktable(sys.argv[1:])

File: swig/perl/mktable.pl
#! /usr/bin/perl -w
#
use greendb;
use strict;

sub mktable {
	my $env = greendb::new_GreenEnv(".");
	greendb::GreenEnv_open($env);
  my $table_name = $ARGV[0];
  print "creating " . $table_name . "\n";
	my $table = greendb::new_Table($table_name, $env);
	my $schema = greendb::Table_get_schema($table);

  my $size = greendb::Schema_size($schema);
  if ($size > 0) {
    print $size . " table already created";
    return;
  }
 
  my $i = 1;
  while ($i < $#ARGV) {
    print "arg ". $i . " " . $ARGV[$i] . "\n";
    my $type = $ARGV[$i];
    $i = $i + 1;
    print "arg ". $i . " " . $ARGV[$i] . "\n";
    my $name = $ARGV[$i];
    $i = $i + 1;
    print "adding ". $type . " " . $name . "\n";
    greendb::Schema_add_column($schema, $name, $type, 0 );
  }
	greendb::Table_close($table);
	greendb::GreenEnv_close($env);
}

mktable();
